src/Build/BackEnd/Components/RequestBuilder/TargetBuilder.cs

```diff
@@ -812,5 +806,37 @@ private void ComputeAfterTargetFailures(string[] targetNames)
                 }
             }
         }
+
+        private bool HasCircularDependenceInTargets(TargetEntry parentTargetEntry, TargetSpecification targetSpecification, out List<string> circularDependenceChain)
+        {
+            TargetEntry currentParent = parentTargetEntry;
+            circularDependenceChain = new List<string>();
+            bool hasCircularDependence = false;
+
+            while (currentParent != null)
+            {
+                if (String.Equals(currentParent.Name, targetSpecification.TargetName, StringComparison.OrdinalIgnoreCase))
```

Comment:
It's relevant which StringComparison you're using here, right?

Reply:
Yes, correct.